Add `alternate receive` bbe (#5290)
ballerina-platform/ballerina-distribution: examples/alternate-receive/alternate_receive.md

# Alternate receive

> The alternate receive action can be used to receive values from several workers.

Reviewer: We need to elaborate more IMO.

> The alternate receive action can be used receive values from several workers. It waits until either a non-error message, a panic termination status on a closed channel, or closure of all channels. Alternate receive actions sets the first non-error value it encounters as the result.

Reviewer 2: Let's avoid using "either" when referring to 3 things.
```suggestion
The alternate receive action can be used to receive values from several workers. It operates by waiting until it encounters a non-error message, a panic termination status on a closed channel, or the closure of all channels. Alternate receive action sets the first non-error value it encounters as the outcome.
```

> The alternate receive action can be used to receive values from several workers. It operates by waiting until it encounters a non-error message, a panic termination status on a closed channel, or the closure of all channels. Alternate receive action sets the first non-error value it encounters as the outcome.

Reviewer 2: One more thing. `alternate-receive` can be used to receive values from sends within the same worker as well. 
May be
```suggestion
The alternate receive action can be used to receive values from mutiple send actions. It operates by waiting until it encounters a non-error message, a panic termination status on a closed channel, or the closure of all channels. Alternate receive action sets the first non-error value it encounters as the outcome.
```

> The alternate receive action can be used to receive one of multiple values corresponding to multiple send actions. It operates by waiting until it encounters a non-error message, a panic termination status on a closed channel, or the closure of all channels. Alternate receive action sets the first non-error value it encounters as the result.

Reviewer 3: We can also mention that it returns the last received error if all channels return errors.

> The alternate receive action can be used to receive one of multiple values corresponding to multiple send actions. It operates by waiting until it encounters a non-error message, a panic termination status on a closed channel, or the closure of all channels. The alternative receive action sets the first non-error value it encounters as the result. If all the channels return errors, it sets the last received error as the result.

Reviewer: ```suggestion
The alternate receive action can be used to receive one of multiple values corresponding to multiple send actions. It operates by waiting until it encounters a non-error message, a panic termination status on a closed channel, or the closure of all channels. The alternative receive action sets the first non-error value it receives as the result. If all the channels return errors, it sets the last received error as the result.
```

> The alternate receive action can be used to receive one of multiple values corresponding to multiple send actions. It operates by waiting until it encounters a non-error message, a panic termination status on a closed channel, or the closure of all channels. The alternative receive action sets the first non-error value it encounters as the result and, in the case all channels return errors, it sets the last received error as the result.

Reviewer: ```suggestion
The alternate receive action can be used to receive one of multiple values corresponding to multiple send actions. It operates by waiting until it encounters a non-error message, a panic termination status on a closed channel, or the closure of all channels. The alternative receive action sets the first non-error value it encounters as the result. If all the channels return errors, it sets the last received error as the result.
```
